superset/security/manager.py

```diff
@@ -1125,3 +1132,20 @@ def raise_for_dashboard_access(self, dashboard: "Dashboard") -> None:
 
             if not can_access:
                 raise DashboardAccessDeniedError()
+
+    # pylint: disable=no-self-use
+    def can_access_based_on_dashboard(self, datasource: "BaseDatasource"):
+        from superset import db
+        from superset.dashboards.filters import DashboardFilter
+        from superset.models.slice import Slice
+        from superset.models.dashboard import Dashboard
+
+        TBL = ConnectorRegistry.sources["table"]
```

Comment:
won't this still break if you pass in a Druid Datasource? you'll be trying to find a table with the same id as the druid datasource